### DISCUSSION TITLE
Something went wrong during completion. Reason: backend not found

### DISCUSSION THREAD

**Discussion by GNBoost0** [mudler/LocalAI#3684] [Q&A]
Hello everyone! :raising_hand:

First of all, I'd like to thank this community and more largely the entire Github community for the incredible work you make available to the world. For me, it's wonderful to see this spirit of sharing access to common creation and development! =)
(This is my first post on Github! I'm moved! Forgive me :stuck_out_tongue_winking_eye:)

Now to my problem!

I installed LocalAI and Chatgpt Telegram Bot following the method described in the ReadMe of @mudler 's deposit at this address: “ https://github.com/mudler/LocalAI/tree/master/examples/telegram-bot ”

I should point out that I installed everything on my external hard drive, the aim being to be able to launch Local AI from anywhere.

Being on Windows 10, I still used different functions for copying and moving files ( copy instead of cp and move instead of mv ). I didn't use the “vim config/config.yml” command, I manually modified the config.yml file to add my Telegram bot token.

I ran the command “docker-compose --env-file config/config.env up --build“.

The images were apparently downloaded correctly, however, when I send a message to my bot on Telegram, I receive this message in response:

> Something went wrong during completion. Reason: backend not found: /tmp/localai/backend_data/backend-assets/grpc/gpt4all-j {“error”:{“code”:500, “message”: “backend not found: /tmp/localai/backend_data/backend-assets/grpc/gpt4all-j”, “type”:“”}} 500 {'error': {'code': 500, 'message': 'backend not found: /tmp/localai/backend_data/backend-assets/grpc/gpt4all-j', 'type': ''}} <CIMultiDictProxy('Date': 'Sat, 28 Sep 2024 10:32:01 GMT', 'Content-Type': 'application/json', 'Content-Length': '119')>

Here the screenshot of the Telegram bot :

![ScreenTelegramBot](https://github.com/user-attachments/assets/23b5fddd-986c-4bfe-b140-f2418da00dcd)

The model is obviously loaded correctly but not its backend.
I did a full search for a file named “gpt4all-j” but without success.

It may not be necessary, but for the sake of clarity I've installed the latest version of Docker, but I've changed the location of its AppData folder on my external hard drive.

I've attached a text file named “Log cmd.exe .txt”. It contains all the command prompt lines from installation to bot response.
Here a screenshot of last lines :

![Screencmd exe](https://github.com/user-attachments/assets/b44b9b03-0535-41f1-8069-10f2774c44d0)

I've also attached the complete folder structure in my external hard drive containing the Docker AppData folder and the complete LocalAI Folder ( E:\LocalAI ). I also put a file containing only the structure of the “chatgpt_telegram_bot” folder ( “E:\LocalAI\LocalAI\examples\telegram-bot\chatgpt_telegram_bot” ).

I apologize for the size of the files, I wanted to be as precise as possible...

I sincerely hope that the problem is due to negligence on my part and that you'll easily find the solution! :blush:

Thanks in advance! :heart:

[chatgpt_telegram_bot_structure.txt](https://github.com/user-attachments/files/17174335/chatgpt_telegram_bot_structure.txt)
[LocalAI_structure.txt](https://github.com/user-attachments/files/17174336/LocalAI_structure.txt)
[Log cmd.exe .txt](https://github.com/user-attachments/files/17174349/Log.cmd.exe.txt)

**Comment by GNBoost0** [comment]
Just a friendly reminder! :pray:

I know everyone has their own things to do, but I hope you'll take the time to help me! :heartpulse:

**_Edit :_** 
Since the day of my post I see you working like a mad @mudler haha! New actions every day! :clap:

Thanks for your incredible work! 
I'll kindly wait and download everything again when you're done! With a bit of luck my problem will no longer exist :stuck_out_tongue_winking_eye:

**Comment by neoacevedo** [comment]
I come here exactly for the same. 

I'm just trying LocalAI from the binary. I installed 2 models type stable diffusion. When I type in the prompt the image, a basic image I want to generate I receive the same error. Trying via curl I get the same result. The error is for both models.

The models are in another drive inside my PC, the models are stored in a hard drive and the binary is in the SSD.

`{"error":{"code":500,"message":"failed to load model with internal loader: backend not found: /tmp/localai/backend_data/backend-assets/grpc/stablediffusion","type":""}}`

**Reply by harnalashok** [reply]
Here is my docker-compose file. I am having the same issue:
Error building Component OpenAI: Error code: 500 - {'error': {'code': 500, 'message': 'failed to load model with internal loader: backend not found: /tmp/localai/backend_data/backend-assets/grpc/gpt4all-j', 'type': ''}} 

(I have used all sorts of tricks. Even installed local-ai without docker using bash script But the message persists.)


version: '3.6'

services:
  api:
    image: quay.io/go-skynet/local-ai:master-ffmpeg-core
    # As initially LocalAI will download the models defined in PRELOAD_MODELS
    # you might need to tweak the healthcheck values here according to your network connection.
    # Here we give a timespan of 20m to download all the required files.
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8080/readyz"]
      interval: 1m
      timeout: 20m
      retries: 20
    ports:
      - 8080:8080
    environment:
      - DEBUG=true
      - MODELS_PATH=/models
      # You can preload different models here as well.
      # See: https://github.com/go-skynet/model-gallery
      - 'PRELOAD_MODELS=[{"url": "github:go-skynet/model-gallery/gpt4all-j.yaml", "name": "gpt-3.5-turbo"}]'
    volumes:
      - ./models:/models:cached
    command: ["/usr/bin/local-ai" ]
  chatgpt:
    depends_on:
      api:
        condition: service_healthy
    image: ghcr.io/mckaywrigley/chatbot-ui:main
    ports:
      - 3000:3000
    environment:
      - 'OPENAI_API_KEY=sk-XXXXXXXXXXXXXXXXXXXX'
      - 'OPENAI_API_HOST=http://api:8080'
